org-eclipse-che-jdt-ls-extension/jdt.ls.extension.api/src/main/java/org/eclipse/che/jdt/ls/extension/api/dto/CheWorkspaceEdit.java

```diff
@@ -10,162 +10,28 @@
  */
 package org.eclipse.che.jdt.ls.extension.api.dto;
 
-import java.util.LinkedHashMap;
 import java.util.LinkedList;
 import java.util.List;
-import java.util.Map;
-import org.eclipse.lsp4j.ResourceChange;
-import org.eclipse.lsp4j.TextDocumentEdit;
-import org.eclipse.lsp4j.TextEdit;
-import org.eclipse.xtext.xbase.lib.Pure;
-import org.eclipse.xtext.xbase.lib.util.ToStringBuilder;
+import org.eclipse.lsp4j.WorkspaceEdit;
 
 /**
- * It is temporary replacement of {@link org.eclipse.lsp4j.WorkspaceEdit} for transporting changes
- * of the refactoring operation.
+ * A workspace edit represents resource changes.
  *
- * <p>A workspace edit represents changes to many resources managed in the workspace. The edit
- * should either provide `changes` or `documentChanges`. If documentChanges are present they are
- * preferred over `changes` if the client can handle versioned document edits.
+ * @author Valeriy Svydenko
  */
-@SuppressWarnings("all")
-public class CheWorkspaceEdit {
-  /** Holds changes to existing resources. */
-  private Map<String, List<TextEdit>> changes;
-
-  /**
-   * An array of `TextDocumentEdit`s to express changes to specific a specific version of a text
-   * document. Whether a client supports versioned document edits is expressed via
-   * `WorkspaceClientCapabilities.versionedWorkspaceEdit`.
-   */
-  private List<TextDocumentEdit> documentChanges;
-
+public class CheWorkspaceEdit extends WorkspaceEdit {
```

Comment:
What's the reasoning for extending WorkspaceEdit? Aren' we inheriting stuff that make no sense? getResourceChanges, for example?

Reply:
I did it to be more approximate to current model of rename command which has to return `WorkspaceEdit` object